### ISSUE TITLE
Stateful Detection of Black-Box Adversarial Attacks

### ISSUE THREAD

**Issue by twrdyyy** [Trusted-AI/adversarial-robustness-toolbox#1373]
**Algorithm described here**
https://arxiv.org/pdf/1907.05587.pdf

**Implementation**
As a part of engineering thesis I'm implementing this algorithm. In addition, I will provide example usages of new module including: jupyter notebook and kubeflow pipeline.

**Comment by twrdyyy** [comment]
@beat-buesser I will soon open pull request with new module

**Comment by twrdyyy** [comment]
@beat-buesser 
PR probably will be ready by the end of this week, it will include:
1. new black-box detection module
  * BlackBoxDetector
  * MemoryQueue class
  * NearestNeighborsWrapper for sklearn lib
  * Implementation of loss functions mentioned in paper (it is helpful for encoder training)
2. implementation of encoder for pytorch (it was missing so I've decided to implement it as well)
3. tests for black-box module 
4. detection example for mnist in pytorch.
---
I am going to use ART and especially this module to implement simple pipeline. I can share with you link to repository as well to include this in examples. Pipeline will use Tensorflow and Beam with kubeflow as UI.

**Comment by twrdyyy** [comment]
@beat-buesser 
To which branch to create a pull request? To dev_1.9.0?

**Comment by beat-buesser** [comment]
Yes, `dev_1.9.0` is the currently dev branch for the next release.

**Comment by twrdyyy** [comment]
@beat-buesser implementation is ready I will create the PR so that you can review changes and propose some improvement
